### PR TITLE
Additional API docs for traces

### PR DESCRIPTION
**Pull request by jander-msft**

### CODE REVIEW

File: documentation/api/definitions.md [hunk 1]
```diff
@@ -50,6 +87,17 @@ The `uid` property is useful for uniquely identifying a process when it is runni
...
 
+## TraceProfile
+
+Enumeration that describes the type of diagnostic trace to capture. Each profile represents a list of event providers, event levels, and keywords.
+
+| Name | Description |
+|---|---|
+| `Cpu` | Tracks CPU usage and general .NET runtime information. |
+| `Http` | Tracks ASP[]().NET request handling and HttpClient requests. |
+| `Logs` | Tracks log events emitted at the `Debug` [log level](https://docs.microsoft.com/dotnet/api/system.diagnostics.tracing.eventlevel) or higher. |
```
Comment (shirhatti) [inline documentation/api/definitions.md:98]: Is this debug or higher? Or does this default to UseAppFilters?

File: documentation/api/trace-custom.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+| 429 Too Many Requests | | There are too many trace requests at this time. Try to request a trace at a later time. | |
+
+> **NOTE:** After the expiration of the trace duration, completing the request may take a long time (up to several minutes) for large applications if `EventProvidersConfiguration.RequestRundown` is set to `true`. The runtime needs to send over the type cache for all managed code that was captured in the trace, known as rundown events. Thus, the length of time of the request may take significantly longer than the requested duration.
+
+## Examples
+
+### Sample Request
+
+```http
+POST https://localhost:52323/trace/21632?durationSeconds=60
```
Comment (shirhatti) [inline documentation/api/trace-custom.md:67]:
```suggestion
POST /trace/21632?durationSeconds=60 HTTP/1.1
Host: localhost:52323/
```

I thought specifying the FQDN in the first line was against the HTTP spec? Or at least an esoteric way to do it

File: documentation/api/trace-custom.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+Authorization: MonitorApiKey QmFzZTY0RW5jb2RlZERvdG5ldE1vbml0b3JBcGlLZXk=
+Content-Type: application/json
+Content:
```
Comment (shirhatti) [inline documentation/api/trace-custom.md:70]:
```suggestion
```

File: documentation/api/definitions.md (hunk 1 above)
Comment (jander-msft) [reply documentation/api/definitions.md:98]: I had to double check it, but the /trace?profile=logs call just uses the Debug level.

This is probably something we want to change and allow it to be overridden with a logLevel query parameter.

File: documentation/api/trace-custom.md (hunk 2 above)
Comment (jander-msft) [reply documentation/api/trace-custom.md:67]: Sure, I can go through everything and reformat it this way.

Comment (jander-msft) [reply documentation/api/trace-custom.md]: I'll fix up the dump ones in the other PR.

File: documentation/api/definitions.md (hunk 1 above)
Comment (jander-msft) [reply documentation/api/definitions.md:109]: Created #183 to track this

File: documentation/api/definitions.md [hunk 4]
```diff
@@ -61,6 +98,17 @@ The `uid` property is useful for uniquely identifying a process when it is runni
...
+| `Metrics` | Tracks [event counters](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/available-counters) from the `System.Runtime`, `Microsoft.AspNetCore.Hosting`, and `Grpc.AspNetCore.Server` event sources. |
```
Comment (wiktork) [inline documentation/api/definitions.md:110]: Did you want to do an example query with profiles?

Comment (wiktork) [reply documentation/api/definitions.md:110]: nvm i see that below.

File: documentation/api/definitions.md [hunk 5]
```diff
@@ -11,6 +11,43 @@ Enumeration that describes the type of information to capture in a managed dump.
...
+| Name | Type | Description |
+|---|---|---|
+| Name | string | The name of the provider from which to capture events. See [Well-known Event Providers](https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers) for commonly used event providers. |
+| Keywords | string | Keyword flags used to enable groups of events. Keyword flags are provider specific. May be specified as a 'stringified' integer or a hexadecimal-encoded integer starting with `0x`. |
+| EventLevel | [EventLevel](https://docs.microsoft.com/dotnet/api/system.diagnostics.tracing.eventlevel) | The level of the events to collect. |
+| Arguments | map (of string) | Additional arguments to the event provider. Names and values are event provider specific. |
+
+## EventProvidersConfiguration
+
+Object describing the list of event providers, keywords, event levels, and additional parameters for capturing a trace.
```
Comment (wiktork) [inline documentation/api/definitions.md:27]: Did you want to link to the clr provider docs from here? (e.g. https://docs.microsoft.com/en-us/dotnet/framework/performance/clr-etw-events)

**Review by wiktork — APPROVED** [review]

File: documentation/api/definitions.md (hunk 5 above)
Comment (jander-msft) [reply documentation/api/definitions.md:27]: In the `EventProvider` section, I already link to https://docs.microsoft.com/dotnet/core/diagnostics/well-known-event-providers, which is more specific to .NET Core.

**Pull request merged**